durabletask/client.py

```diff
@@ -96,8 +98,25 @@ def __init__(self, *,
                  metadata: Optional[list[tuple[str, str]]] = None,
                  log_handler: Optional[logging.Handler] = None,
                  log_formatter: Optional[logging.Formatter] = None,
-                 secure_channel: bool = False):
-        channel = shared.get_grpc_channel(host_address, metadata, secure_channel=secure_channel)
+                 secure_channel: bool = False,
+                 interceptors: Optional[list[DefaultClientInterceptorImpl]] = None):
```

Comment:
This doesn't look right. `DefaultClientInterceptorImpl` is an internal class that we own and shouldn't be exposing to our users. Rather, there should be some base class specified here, like `grpc.UnaryUnaryClientInterceptor` or `grpc.UnaryStreamClientInterceptor`, that customers can derive their own interceptor types from. So this should be something like:
```suggestion
                 interceptors: Optional[list[Union[grpc.UnaryUnaryClientInterceptor, grpc.UnaryStreamClientInterceptor]]] = None):
```